tests/unit_tests/cuda/index_geometry_cuda_kernel.cu

```diff
@@ -0,0 +1,45 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <vecmem/containers/device_vector.hpp>
+
+#include "definitions/cuda_defs.hpp"
+#include "index_geometry_cuda_kernel.hpp"
+
+namespace detray {
+
+__global__ void index_geometry_test_kernel(
+    index_geometry_data<geometry> geometry_data,
+    vecmem::data::vector_view<typename geometry::volume_type> output_data) {
+
+    index_geometry<vecmem::device_vector> g(geometry_data);
+
+    vecmem::device_vector<typename geometry::volume_type> output_device(
+        output_data);
+
+    for (unsigned int i = 0; i < g.n_volumes(); i++) {
```

Comment:
If we are just copying volumes, do we need distinct kernels for the two geometry types?

Reply:
I see, lets keep it